packages/ui/src/hooks/useIconElement.tsx

```diff
@@ -0,0 +1,21 @@
+import { Icon } from '..';
+import React from 'react';
+import type { IconifyIcon } from '@iconify/types';
+
+export type IconTypeProp = string | IconifyIcon;
```

Comment:
Add `export type IconTypeProp = string | IconifyIcon;` on `Icon.tsx` file. I think `IconType` is better.